apps/autoscheduler/autotune_loop.sh

```diff
@@ -107,19 +119,43 @@ for ((i=$((FIRST+1));i<1000000;i++)); do
     # Copy the weights being used into the batch folder so that we can repro failures
     mkdir -p ${DIR}
     cp weights/* ${SAMPLES}/batch_${i}/
-    
+
     echo Compiling ${BATCH_SIZE} samples for batch_${i}...
     for ((b=0;b<${BATCH_SIZE};b++)); do
         S=$(printf "%d%02d" $i $b)
         FNAME=$(printf "%s_batch_%02d_sample_%02d" ${PIPELINE} $i $b)
-        make_sample "${DIR}/${b}" $S $FNAME &
+        make_sample_lib "${DIR}/${b}" $S $FNAME &
     done
     wait
 
+    echo Compiling ${BATCH_SIZE} stubs for batch_${i}...
```

Comment:
Does this encapsulate failures in the same way as the old script? If anything went wrong compiling any one sample in the old script, it didn't affect other samples. I could imagine this bringing down the whole batch.

Reply:
Yes, this definitely would bring down the whole batch. But how would that have worked before? ie, if one sample failed to compile, what happened when the script tried to execute it -- wouldn't it fail then?

Reply:
Yes, and it results in a truncated sample file, which train_cost_model just rejects at training time.

Reply:
(rejects gracefully - it skips it)

Reply:
Sounds like I should try injecting deliberate failures to see what happens and mimic the old behavior.

Reply:
> If anything went wrong compiling any one sample in the old script, it didn't affect other samples. 

How likely is this in practice, and/or when do you expect it to happen? (Getting Bazel to fail gracefully in this situation is going to require some thought.)

Reply:
The random pipeline generator doesn't always generate valid pipelines, and a random schedule on a pipeline might not be valid due to a corner case bug in the new autoscheduler. Both of these are fairly frequent.

Reply:
Ouch -- that will make my favored technique harder to make work, because reasons. Better to find out now, though.